File: MPP_MusicFestival_Java/src/main/java/repository/TicketDBRepository.java
package repository;

import domain.Artist;
import domain.Ticket;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class TicketDBRepository implements TicketRepository {

    private JdbcUtils dbUtils;
    private static final Logger logger = LogManager.getLogger();

    private ShowRepository showRepository;
    private OfficeEmployeeRepository officeEmployeeRepository;

    public TicketDBRepository(Properties props, ShowRepository showRepository, OfficeEmployeeRepository officeEmployeeRepository) {
        logger.info("Initializing TicketDBRepository with properties: {} ", props);
        dbUtils = new JdbcUtils(props);
        this.showRepository = showRepository;
        this.officeEmployeeRepository = officeEmployeeRepository;
    }


    @Override
    public void save(Ticket entity) {
        logger.traceEntry("saving ticket {} ", entity);
        Connection con = dbUtils.getConnection();

        try (PreparedStatement preStmt = con.prepareStatement("insert into Ticket(show_id, employee_id, buyer_name) values (?,?,?)")) {
            preStmt.setLong(1, entity.getShowId());
            preStmt.setLong(2, entity.getEmployeeId());
            preStmt.setString(3, entity.getBuyerName());
            preStmt.executeUpdate();
        } catch (SQLException ex) {
            logger.error(ex);
            System.out.println("Error DB " + ex);
        }
        logger.traceExit();
    }

    @Override
    public void delete(Long aLong) {
        logger.traceEntry("deleting ticket with {}", aLong);
        Connection con = dbUtils.getConnection();

        try (PreparedStatement preStmt = con.prepareStatement("delete from Ticket where id=?")) {
            preStmt.setLong(1, aLong);
            preStmt.executeUpdate();
        } catch (SQLException ex) {
            logger.error(ex);
            System.out.println("Error DB " + ex);
        }
        logger.traceExit();
    }

    @Override
    public void update(Long aLong, Ticket entity) {
    }

    @Override
    public Ticket findOne(Long aLong) {
        logger.traceEntry("finding ticket with id {} ", aLong);
        Connection con = dbUtils.getConnection();

        try (PreparedStatement preStmt = con.prepareStatement("select * from Ticket where id=?")) {
            preStmt.setLong(1, aLong);
            try (ResultSet result = preStmt.executeQuery()) {
                if (result.next()) {
                    long id = result.getLong("id");
                    long show_id = result.getLong("show_id");
                    long employee_id = result.getLong("employee_id");
                    String buyer_name = result.getString("buyer_name");

                    Ticket ticket = new Ticket(id, show_id, employee_id, buyer_name);
                    logger.traceExit(ticket);
                    return ticket;
                }
            }
        } catch (SQLException ex) {
            logger.error(ex);
            System.out.println("Error DB " + ex);
        }
        logger.traceExit("No ticket found with id {}", aLong);
        return null;
    }

    @Override
    public List<Ticket> findAll() {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        List<Ticket> tickets = new ArrayList<>();
        try (PreparedStatement preStmt = con.prepareStatement("select * from Ticket")) {
            try (ResultSet result = preStmt.executeQuery()) {
                while (result.next()) {
                    long id = result.getLong("id");
                    long show_id = result.getLong("show_id");
                    long employee_id = result.getLong("employee_id");
                    String buyer_name = result.getString("buyer_name");

                    Ticket ticket = new Ticket(id, show_id, employee_id, buyer_name);
                    tickets.add(ticket);
                }
            }

        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
        return tickets;
    }
}

File: MusicFestival/Repository/TicketDBRepository.cs
using log4net;
using MPP_MusicFestival.Domain;
using MPP_MusicFestival.Repository;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicFestival.Repository
{


    class TicketDBRepository : TicketRepository
    {

        private static readonly ILog log = LogManager.GetLogger("ShowDBRepository");

        IDictionary<String, string> props;

        private ShowRepository showRepository;
        private OfficeEmployeeRepository officeEmployeeRepository;

        public TicketDBRepository(IDictionary<String, string> props, ShowRepository showRepository, OfficeEmployeeRepository officeEmployeeRepository)
        {
            log.Info("Creating TicketDBRepository ");
            this.props = props;
            this.showRepository = showRepository;
            this.officeEmployeeRepository = officeEmployeeRepository;
        }

        public void delete(long id)
        {
            IDbConnection con = DBUtils.getConnection(props);
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "delete from Ticket where id=@id";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);
                var dataR = comm.ExecuteNonQuery();
                //if (dataR == 0)
                //  throw new RepositoryException("No task deleted!");
            }
        }

        public IEnumerable<Ticket> findAll()
        {
            IDbConnection con = DBUtils.getConnection(props);
            IList<Ticket> tickets = new List<Ticket>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Ticket";

                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        long id = dataR.GetInt32(0);
                        long show_id = dataR.GetInt32(1);
                        long employee_id = dataR.GetInt32(2);
                        String buyer_name = dataR.GetString(3);

                        Ticket ticket = new Ticket(id, show_id, employee_id, buyer_name);
                        tickets.Add(ticket);
                    }
                }
            }

            return tickets;
        }

        public Ticket findOne(long id)
        {
            log.InfoFormat("Entering findOne with value {0}", id);
            IDbConnection con = DBUtils.getConnection(props);

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Ticket where id=@id";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        long idT = dataR.GetInt32(0);
                        long show_id = dataR.GetInt32(1);
                        long employee_id = dataR.GetInt32(2);
                        String buyer_name = dataR.GetString(3);

                        Ticket ticket = new Ticket(idT, show_id, employee_id, buyer_name);

                        log.InfoFormat("Exiting findOne with value {0}", ticket);
                        return ticket;
                    }
                }
            }
            log.InfoFormat("Exiting findOne with value {0}", null);
            return null;
        }

        public void save(Ticket entity)
        {
            var con = DBUtils.getConnection(props);

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "insert into Ticket(show_id, employee_id, buyer_name) values (@show_id, @employee_id, @buyer_name)";
                var paramSId = comm.CreateParameter();
                paramSId.ParameterName = "@show_id";
                paramSId.Value = entity.showId;
                comm.Parameters.Add(paramSId);

                var paramEId = comm.CreateParameter();
                paramEId.ParameterName = "@employee_id";
                paramEId.Value = entity.employeeId;
                comm.Parameters.Add(paramEId);

                var paramBName = comm.CreateParameter();
                paramBName.ParameterName = "@buyer_name";
                paramBName.Value = entity.buyerName;
                comm.Parameters.Add(paramBName);


                var result = comm.ExecuteNonQuery();
                //if (result == 0)
                //  throw new RepositoryException("No task added !");
            }
        }

        public void update(long id, Ticket entity)
        {
            throw new NotImplementedException();
        }
    }
}
